File: 프로그래머스/0/120897. 약수 구하기/약수 구하기.js
function solution(n) {
    var answer = [];
    
    for(let i = 1 ; i <= Math.sqrt(n) ; i++)
        if(n % i === 0){
            answer.push(i);
            if(i !== parseInt(n / i))
                answer.push(parseInt(n / i));
        }
    
    answer.sort((a, b) => a - b)
    
    return answer;
}

File: 프로그래머스/0/120897. 약수 구하기/약수 구하기.java
import java.util.ArrayList;
import java.util.List;

class Solution {
    public int[] solution(int n) {
        
        List<Integer> list = new ArrayList<>();
        
        for(int i = 1; i <= (int)Math.sqrt(n) ; i++)
            if(n % i == 0){
                list.add(i);
                if(n / i != i)
                    list.add(n / i);
            }
        
        return list.stream()
            .sorted()
            .mapToInt(Integer::valueOf)
            .toArray();
    }
}